build.gradle

```diff
@@ -25,23 +27,21 @@ subprojects {
     apply plugin: "com.google.osdetector"
     // The plugin only has an effect if a signature is specified
     apply plugin: "ru.vyarus.animalsniffer"
-    // jdk10 not supported by errorprone: https://github.com/google/error-prone/issues/860
-    if (!JavaVersion.current().isJava10Compatible() &&
-        rootProject.properties.get('errorProne', true)) {
-        apply plugin: "net.ltgt.errorprone"
 
+    apply plugin: "net.ltgt.errorprone"
+    if (rootProject.properties.get('errorProne', true)) {
         dependencies {
-            // The ErrorProne plugin defaults to the latest, which would break our
-            // build if error prone releases a new version with a new check
             errorprone 'com.google.errorprone:error_prone_core:2.3.2'
+            errorproneJavac 'com.google.errorprone:javac:9+181-r4173-1'
```

Comment:
Does this mean we'd always be using Java 9 for compilation? Or does it run javac twice: once for the real build and another for errorprone?

Reply:
This javac 9 is only used when using JDK 8; this is the same behavior actually as with the old net.ltgt.errorprone plugin.

With JDK 9 and above, the JDK's javac will be used, and this dependency won't be used at all.